portal-api/src/external-api/bank-holidays.js

```diff
@@ -0,0 +1,34 @@
+const axios = require('axios');
+const dotenv = require('dotenv');
+
+dotenv.config();
+const { EXTERNAL_API_URL, EXTERNAL_API_KEY } = process.env;
+
+const headers = {
+  'Content-Type': 'application/json',
+  'x-api-key': EXTERNAL_API_KEY,
+};
+
+/**
+ * Fetches details of UK bank holidays from external-api.
+ * @returns {Promise<object>}
+ */
+const getBankHolidays = async () => {
+  const response = await axios.get(`${EXTERNAL_API_URL}/bank-holidays`, { headers });
+  return response.data;
+};
+
+/**
+ * Fetches a list of bank holiday dates for the specified UK region
+ * @param {'england-and-wales' | 'scotland' | 'northern-ireland'} region
+ * @returns {Promise<Date[]>}
+ */
+const getBankHolidayDatesForRegion = async (region) => {
+  const allUkBankHolidays = await getBankHolidays();
+  return allUkBankHolidays[region].events.map((event) => new Date(event.date));
+};
+
+module.exports = {
+  getBankHolidays,
+  getBankHolidaysForRegion: getBankHolidayDatesForRegion,
```

Comment:
Assume this was a result of renaming the function but we probably want to rename it here too?

Reply:
Exactly that. Thanks for the catch!